_posts/2019-09-02-shorter-isn't-better.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: post
+title: "Shorter functions are't always better"
+date: 2019-09-05 10:00:00 +0800
+categories: blog
+author: Lex Huang
+---
+
+# Shorter functions aren't always better
+
+This is just a thought that came to me during day to day code reviews and I want to share the idea with you. My point is that for functions, readability, understandability and maintainability is much more important the length and the abstraction level.
+
+## A Trivial Example
+
+Consider you want to implement a simple algorism that show the balance in foreign currency given a list of incomes and outcomes. I will do this in `Javascript`. For the sake of simplicity, I will **ignore** the `float point issue` of ECMAScript and `type/value validation` for the arguments.
+
+So the algorism needs 3 arguments: a balance, list of numbers as income in positive number and outcome in negative number, and an exchange rate. And the result is a number.
```

Comment:
The `algorithm` needs 3 arguments: the balance, the list of numbers containing incomes with positive values or expenditures with negative values, and the exchange rate, and return a number value.